lib/dialects/mssql/query-generator.js

```diff
@@ -322,7 +322,7 @@ const QueryGenerator = {
     return sql;
   },
 
-  upsertQuery(tableName, insertValues, updateValues, where, rawAttributes) {
+  upsertQuery(tableName, insertValues, updateValues, where, model) {
```

Comment:
@felixfbecker I've used some internals myself. I will make a PR to expose them properly later on. But I'm always ready for it to blow up in my face. Anything that isn't documented is subject to change without notice, the first rule of open source 😎 😎. we gotta draw the line somewhere right? 😅 😅 Or either the code gets complicated or no good change can make it to the end.